### PR TITLE
#54 - Added Manifest.layers() method

### PR DESCRIPTION
**Pull request by olegmoz**
Part of #54 
Added Manifest.layers() method to read all manifest layer digests. This is required to check their presence when manifest is added.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#111](https://github.com/artipie/docker-adapter/issues/111) is assigned to @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/45bc2f9b-7d7e-42b4-b3a0-e8f78973dee9); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/main/java/com/artipie/docker/manifest/JsonManifest.java [hunk 1]
```diff
@@ -68,8 +72,27 @@ public JsonManifest(final Content source) {
         );
     }
 
+    @Override
+    public CompletionStage<Collection<Digest>> layers() {
+        return this.json().thenApply(
+            root -> root.getJsonArray("layers").getValuesAs(JsonValue::asJsonObject).stream()
+                .map(obj -> obj.getString("digest"))
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/docker/manifest/JsonManifest.java:79]: @olegmoz I'd suggest to rename `obj` into `digest` to make it more verbose

File: src/test/java/com/artipie/docker/manifest/JsonManifestTest.java [hunk 2]
```diff
@@ -79,6 +82,21 @@ void shouldFailConvertToUnknownType() {
...
+        final JsonManifest manifest = new JsonManifest(
+            new Content.From(
+                "{\"layers\":[{\"digest\":\"sha256:123\"},{\"digest\":\"sha256:abc\"}]}".getBytes()
+            )
+        );
+        MatcherAssert.assertThat(
+            manifest.layers().toCompletableFuture().join().stream()
+                .map(Digest::string)
+                .collect(Collectors.toList()),
+            Matchers.containsInAnyOrder("sha256:123", "sha256:abc")
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/docker/manifest/JsonManifestTest.java:96]: @olegmoz let's extract digests into local variables and use them to construct content json and in assertion

File: src/main/java/com/artipie/docker/manifest/JsonManifest.java (hunk 1 above)
Comment (olegmoz) [reply src/main/java/com/artipie/docker/manifest/JsonManifest.java:79]: @olenagerasimova changed it to `layer`, I think it makes more sense

File: src/test/java/com/artipie/docker/manifest/JsonManifestTest.java (hunk 2 above)
Comment (olegmoz) [reply src/test/java/com/artipie/docker/manifest/JsonManifestTest.java:96]: @olenagerasimova fixed

**Comment by 0crat** [comment]
@olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by olenagerasimova** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/artipie/docker-adapter/issues/111#issuecomment-630237032))

@olenagerasimova This is what I know about this job in CT2E6TK9B, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#111](https://github.com/artipie/docker-adapter/issues/111) is [in scope](http://www.zerocracy.com/policy.html#14) for 5days
  * The role is `REV`
  * The job is assigned to @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova) for [5days](http://www.zerocracy.com/policy.html#8)
  * There is no monetary reward attached, it's a [free](http://www.zerocracy.com/policy.html#2) job
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @olegmoz[/z](https://www.0crat.com/u/olegmoz): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/CT2E6TK9B?q=%7Bjob%3A%27gh%3Aartipie%2Fdocker-adapter%23111%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by g4s8** [comment]
@0crat out

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@olenagerasimova please make sure you will find at least 3 issues during next CR, thanks!

**Comment by sereshqua** [comment]
@olenagerasimova ping

**Comment by olenagerasimova** [comment]
@sereshqua I'll do my best, sorry for the delay

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**